pkg/controller/elasticsearch/certificates/transport/reconcile_test.go

```diff
@@ -188,13 +191,49 @@ func TestReconcileTransportCertificatesSecrets(t *testing.T) {
 				assert.NotContains(t, transportCerts2.Data, "test-es-name-es-sset2-2.tls.key")
 			},
 		},
+		{
+			name: "can disable certificate reconciliation",
+			args: args{
+				ca: testRSACA,
+				es: newEsBuilder().addNodeSet("sset1", 2).disableTransportCerts().build(),
+				initialObjects: []client.Object{
+					// one pod pending update to add the marker annotation
+					newPodBuilder().forEs(testEsName).inNodeSet("sset1").withIndex(0).withIP("1.1.1.2").build(),
+					// one pod rotated with the marker annotation
+					newPodBuilder().forEs(testEsName).inNodeSet("sset1").withIndex(1).withIP("1.1.1.2").withAnnotations(map[string]string{
+						esv1.TransportCertDisabledAnnotationName: "true",
+					}).build(),
+					newtransportCertsSecretBuilder(testEsName, "sset1").forPodIndices(0, 1).build(),
+				},
+			},
+			wantRequeue: true, // one pod does not have the annotation yet so requeue is expected
+			wantErr:     false,
+			assertSecrets: func(t *testing.T, secrets corev1.SecretList) {
+				t.Helper()
+				// Check that there is 1 Secret per StatefulSet
+				assert.Equal(t, 1, len(secrets.Items))
+
+				transportCerts1 := getSecret(secrets, "test-es-name-es-sset1-es-transport-certs")
+				assert.NotNil(t, transportCerts1)
+				// Transport certs CA is still there
+				assert.Contains(t, transportCerts1.Data, "ca.crt")
+				// disabled marker is there
+				assert.Contains(t, transportCerts1.Data, "transport.certs.disabled")
```

Comment:
I think you can use the constant here.

Reply:
I try to avoid to use constants in tests, otherwise defintion and check are the same thing and nothing is tested. Not a big deal here as the value is effectively arbitrary, but as a matter of principle.